### PR TITLE
Fix MDX syntax, broken links, and use of the `observability` variable

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Description

Fixes syntax errors, broken links, and updates based on the updated `observability` variable found while working on https://github.com/elastic/docs/pull/3096.

Screenshot 2024-10-31 at 9 50 42 AM

(This will make it easier to pull in more updates ahead of finishing https://github.com/elastic/docs/pull/3096.)

### Documentation sets edited in this PR

_Check all that apply._

- [ ] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue

N/A

## Checklist

- [x] ~~Product/Engineering Review~~
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [x] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [ ] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

File: docs/en/serverless/logging/troubleshoot-logs.mdx [hunk 1]
```diff
@@ -20,7 +20,7 @@ if you don't have the required privileges to create an API key, you'll see the f
 
 You need to either:
 
-* Ask an administrator to update your user role to at least **Deployment access** → **Admin**. Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles" />. After your use role is updated, restart the onboarding flow.
+* Ask an administrator to update your user role to at least **Deployment access** → **Admin**. Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles">Assign user roles and privileges</DocLink>. After your use role is updated, restart the onboarding flow.
```
Comment (dedemorton) [inline docs/en/serverless/logging/troubleshoot-logs.mdx:23]: Is hard coding the link text really required? I just think this is a maintainability issue. (IMO it's better for the tool to keep the link text up-to-date unless there is a compelling reason to hard code it.)

I haven't looked at the original output so maybe this does solve a problem, but I've used the syntax without the link text in the past, and it's worked.

**Review by dedemorton — APPROVED** [review]
LGTM--just had a question.

I've eyeballed this (did not look at the preview).

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-f28o1kivj-elastic-dev.vercel.app

* Built with commit cd8c2b8621efebe15849e7369896f897afa27763

> Issues? Visit #next-docs in Slack

File: docs/en/serverless/logging/troubleshoot-logs.mdx (hunk 1 above)
Comment (colleenmcginnis) [reply docs/en/serverless/logging/troubleshoot-logs.mdx:23]: This was specifically for migrating to AsciiDoc -- it was complicated to get the link text for a page in a different repo -- but I can put some more time into the script if this is a deal breaker.

File: docs/en/serverless/alerting/view-alerts.mdx [hunk 2]
```diff
@@ -87,7 +87,7 @@ Use the toolbar buttons in the upper-left of the alerts table to customize the c
 
 For example, click **Fields** and choose the `Maintenance Windows` field.
 If an alert was affected by a maintenance window, its identifier appears in the new column.
-For more information about their impact on alert notifications, refer to <DocLink slug="/serverless/maintenance-windows" />.
+For more information about their impact on alert notifications, refer to <DocLink slug="/serverless/maintenance-windows">Maintenance windows</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/alerting/view-alerts.mdx:90]:
```suggestion
For more information about their impact on alert notifications, refer to <DocLink slug="/serverless/maintenance-windows" />.
```

File: docs/en/serverless/logging/troubleshoot-logs.mdx (hunk 1 above)
Comment (colleenmcginnis) [reply docs/en/serverless/logging/troubleshoot-logs.mdx:23]:
```suggestion
* Ask an administrator to update your user role to at least **Deployment access** → **Admin**. Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles" />. After your use role is updated, restart the onboarding flow.
```

File: docs/en/serverless/partials/roles.mdx [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <DocCallOut title="Required role">
-  The **{props.role}** role or higher is required to {props.goal}. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
+  The **{props.role}** role or higher is required to {props.goal}. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles">Assign user roles and privileges</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/partials/roles.mdx:2]:
```suggestion
  The **{props.role}** role or higher is required to {props.goal}. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
```

File: docs/en/serverless/projects/billing.mdx [hunk 4]
```diff
@@ -19,6 +19,6 @@ When you use Elastic Observability, your bill is calculated based on data volume
 Browser (journey) based tests are charged on a per-test-run basis,
 and Ping (lightweight) tests have an all-you-can-use model per location used.
 
-For more information, refer to <DocLink slug="/serverless/general/serverless-billing" />.
+For more information, refer to <DocLink slug="/serverless/general/serverless-billing">Serverless billing dimensions</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/projects/billing.mdx:22]:
```suggestion
For more information, refer to <DocLink slug="/serverless/general/serverless-billing" />.
```

File: docs/en/serverless/quickstarts/k8s-logs-metrics.mdx [hunk 5]
```diff
@@ -16,7 +16,7 @@ The kubectl command installs the standalone Elastic Agent in your Kubernetes clu
 ## Prerequisites
 
 - An ((observability)) project. To learn more, refer to <DocLink slug="/serverless/observability/create-an-observability-project" />.
-- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
+- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles">Assign user roles and privileges</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/quickstarts/k8s-logs-metrics.mdx:19]:
```suggestion
- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
```

File: docs/en/serverless/quickstarts/monitor-hosts-with-elastic-agent.mdx [hunk 6]
```diff
@@ -19,7 +19,7 @@ The script also generates an ((agent)) configuration file that you can use with
 ## Prerequisites
 
 - An ((observability)) project. To learn more, refer to <DocLink slug="/serverless/observability/create-an-observability-project" />.
-- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
+- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles">Assign user roles and privileges</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/quickstarts/monitor-hosts-with-elastic-agent.mdx:22]:
```suggestion
- A user with the **Admin** role or higher—required to onboard system logs and metrics. To learn more, refer to <DocLink slug="/serverless/general/assign-user-roles" />.
```

File: docs/en/serverless/synthetics/synthetics-feature-roles.mdx [hunk 7]
```diff
@@ -42,4 +42,4 @@ requirements and the minimum privileges required to use specific features.
 </DocRow>
 </DocTable>
 
-Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles" />.
+Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles">Assign user roles and privileges</DocLink>.
```
Comment (colleenmcginnis) [inline docs/en/serverless/synthetics/synthetics-feature-roles.mdx:45]:
```suggestion
Read more about user roles in <DocLink slug="/serverless/general/assign-user-roles" />.
```

**Review by colleenmcginnis — COMMENTED** [review]
I'll try to fix these in the script instead.

**Comment by colleenmcginnis** [comment]
@dedemorton this is ready for a second look! (I pushed a commit so it dismissed your earlier approval.)

**Review by dedemorton — APPROVED** [review]
LGTM

**Pull request merged**